pkgs/development/tools/haskell/haskell-language-server/withWrapper.nix

```diff
@@ -27,32 +127,39 @@ let
         '';
       }))
       ((if dynamic then enableCabalFlag else disableCabalFlag) "dynamic")
-    ] ++ optionals (!dynamic) [
+      (if supportedFormatters != [] then removeUnnecessaryFormatters else lib.id)
+    ]
+    ++ lib.optionals (!dynamic) [
       justStaticExecutables
     ]);
+
   targets = version:
     let packages = getPackages version;
-    in [
-      "haskell-language-server-${packages.ghc.version}"
-    ];
+    in [ "haskell-language-server-${packages.ghc.version}" ];
+
   makeSymlinks = version:
-    concatMapStringsSep "\n" (x:
-      "ln -s ${
-        tunedHls (getPackages version)
-      }/bin/haskell-language-server $out/bin/${x}") (targets version);
-in assert supportedGhcVersions != []; stdenv.mkDerivation {
+    lib.concatMapStringsSep "\n"
+      (x:
+        "ln -s ${
+          tunedHls (getPackages version)
+        }/bin/haskell-language-server $out/bin/${x}")
+      (targets version);
+
+in stdenv.mkDerivation {
```

Comment:
I think this could benefit from the `finalAttrs` pattern because this makes the configuration more discoverable, via the REPL for instance and more composable, you could append to the supportedFormatters via overrideAttrs, whereas it's harder/not possible (to my knowledge) with `override`

Reply:
This is out-of-scope for this PR.